### PR TITLE
feature(click outside): remove click outside directive and integrate …

### LINKED ISSUE #84: Convert click-outside directive to Angular 1.5 component
In order to be able to upgrade the header to Angular 2 component we need to get rid of directives and transform them in Angular 1.5 components

### PR DESCRIPTION
**Pull request by dami-gg**
Closes #84

### CODE REVIEW

File: src/solutioncenter.header.controller.js [hunk 1]
```diff
@@ -1,116 +1,172 @@
...
+      $document.bind('touchstart', eventHandler);
+    }
+
+    function unregisterListener() {
+      $document.unbind('click', eventHandler);
+      $document.unbind('touchstart', eventHandler);
+    }
+
+    function eventHandler(event) {
+      var header = angular.element('.solution-center-header');
```
Comment (gregdeane) [inline src/solutioncenter.header.controller.js]: I'm not able to get this to work with jqLite. I know there is a way though. Please look around for the jqLite solution.

File: src/solutioncenter.header.controller.js [hunk 2]
```diff
@@ -1,116 +1,172 @@
...
-            'my.nanorep.com/widget/scripts/embed.js?account=zalandobrands';
-          var _head = document.getElementsByTagName('head')[0];
-          _head.appendChild(sc);
-        })();
-        // added this to generate code so we can run this outside the global scope
-        window._nRepData = _nRepData;
-        /*eslint-enable */
-        vm.helpWidgetLoaded = true;
-      };
+  function ScHeaderController($scope, $window, $attrs, $document) {
```
Comment (gregdeane) [inline src/solutioncenter.header.controller.js]: Nice cleanup!

File: dist/solutioncenter.header.js [hunk 3]
```diff
@@ -58,123 +18,180 @@ angular.module('solutioncenter.header', [])
...
+          if (vm.modules && vm.modules.length > 0 && !vm.helpWidgetLoaded) {
+            loadHelpWidget(vm.modules);
           }
         });
-        return products;
+    }
+
+    /////////////////////
+
+    function loadHelpWidget(modules) {
```
Comment (gregdeane) [inline dist/solutioncenter.header.js]: Let's make sure of SRP when doing the ng2 header. This function inside the controller is ludicrous.

**Comment by gregdeane** [comment]
Please address comments and I'll approve.

File: src/solutioncenter.header.controller.js (hunk 1 above)
Comment (dami-gg) [inline src/solutioncenter.header.controller.js]: this is the solution I've found ;)

angular.element(document.querySelector('.solution-center-header'));

File: dist/solutioncenter.header.js (hunk 3 above)
Comment (dami-gg) [inline dist/solutioncenter.header.js]: I agree but it was too much refactoring this time for a component we're going to stop maintaining soon. I'll do so for the new version

**Comment by dami-gg** [comment]
:+1:

**Comment by dami-gg** [comment]
@gregdeane I've added a new version which though it's not 100% accurate (since it searches for an element of the same tag and same classes, which does not guarantee it's the only one in the whole code), it's quite unlikely to break in our case.

Since I'm going to be on holidays feel free to change it if you see any possible improvement or to merge it if I'm not here

**Comment by dami-gg** [comment]
:+1:

**Comment by gregdeane** [comment]
Looks good man. Demo works, SC integration works, unit tests pass, e2e tests run fine. Great work!

One last thing: It's off topic but please add `data-e2e="sch-bs-toggle-name"` to the following element (highlighted in yellow):

https://github.com/zalando-incubator/solution-center-header/blob/1b30da028cdef52c2848bb71539af0a240d2ae85/src/solutioncenter.header.component.html#L42

After that, plus one's all around. Thx!

**Comment by gregdeane** [comment]
Sorry, also remove jQuery from package.json.

**Comment by cjhowald** [comment]
@gregdeane Fine to remove it. We just need to find another way to do the unit test check.

**Comment by gregdeane** [comment]
@cjhowald Thanks. I am curious about having added jQuery just for unit tests since jqLite is part of Angular. Do you happen to know the story behind that decision?

I understand that jqLite isn't as friendly as jQuery but typically it's possible to use jqLite in its place.

**Comment by dami-gg** [comment]
I've added the data-e2e attribute, let's do the jQuery part in another ticket so we can close this. We're not releasing a new version yet anyway. Could you guys create that issue if you still it makes sense? Thanks

**Comment by dami-gg** [comment]
:+1:

**Comment by cjhowald** [comment]
@gregdeane The unit test checks the existence of elements with particular class name. The `find` method in jqLite only supports matching on element tag names, not class names, which is why jquery was included. Probably there is a way to avoid using jQuery here.

https://docs.angularjs.org/api/ng/function/angular.element

**Comment by gregdeane** [comment]
Thanks for the data-e2e add.

**Comment by gregdeane** [comment]
👍

**Pull request merged**